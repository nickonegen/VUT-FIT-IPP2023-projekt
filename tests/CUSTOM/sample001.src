.IPPcode23
CREATEFRAME      # 1
DEFVAR GF@x      # 2
DEFVAR GF@y      # 3
DEFVAR TF@tmp    # 4
MOVE GF@x int@5  # 5
MOVE GF@y int@2  # 6
ADD TF@tmp GF@x GF@y # 7
POPS GF@x       # 8
JUMP label      # 9
LABEL label	# 10
WRITE GF@x     # 11
EXIT int@0     # 12