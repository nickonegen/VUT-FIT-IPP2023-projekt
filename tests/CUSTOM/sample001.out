<?xml version="1.0" encoding="UTF-8"?>
<program language="IPPcode23"><instruction order="1" opcode="CREATEFRAME"/><instruction order="2" opcode="DEFVAR"><arg1 type="var">GF@x</arg1></instruction><instruction order="3" opcode="DEFVAR"><arg1 type="var">GF@y</arg1></instruction><instruction order="4" opcode="DEFVAR"><arg1 type="var">TF@tmp</arg1></instruction><instruction order="5" opcode="MOVE"><arg1 type="var">GF@x</arg1><arg2 type="int">5</arg2></instruction><instruction order="6" opcode="MOVE"><arg1 type="var">GF@y</arg1><arg2 type="int">2</arg2></instruction><instruction order="7" opcode="ADD"><arg1 type="var">TF@tmp</arg1><arg2 type="var">GF@x</arg2><arg3 type="var">GF@y</arg3></instruction><instruction order="8" opcode="POPS"><arg1 type="var">GF@x</arg1></instruction><instruction order="9" opcode="JUMP"><arg1 type="label">label</arg1></instruction><instruction order="10" opcode="LABEL"><arg1 type="label">label</arg1></instruction><instruction order="11" opcode="WRITE"><arg1 type="var">GF@x</arg1></instruction><instruction order="12" opcode="EXIT"><arg1 type="int">0</arg1></instruction></program>
